### PR TITLE
curl: update and cleanup

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Maintainer: @stangri

### CODE REVIEW

File: net/curl/Makefile [hunk 1]
```diff
@@ -8,14 +8,12 @@
...
-PKG_VERSION:=7.83.1
+PKG_VERSION:=7.84.0
 PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_SOURCE_URL:=https://dl.uxnr.de/mirror/curl/ \
-	https://curl.askapache.com/download/ \
-	https://curl.se/download/
-PKG_HASH:=2cb9c2356e7263a1272fd1435ef7cdebf2cd21400ec287b068396deb705c22c4
+PKG_SOURCE_URL:=https://curl.se/download/
```
Comment (diizzyy) [inline net/curl/Makefile:15]: Default to GitHub and add curl.se as fallback
https://github.com/curl/curl/releases/tag/curl-7_84_0

Comment (stangri) [reply net/curl/Makefile:15]: Great idea, is there a way to convert 7.84.0 to 7_84_0 in the Makefile as to not to hardcode the specific version in the URL?

File: net/curl/Makefile [hunk 2]
```diff
@@ -93,7 +91,7 @@ define Package/libcurl
   CATEGORY:=Libraries
   DEPENDS:= +LIBCURL_WOLFSSL:libwolfssl +LIBCURL_OPENSSL:libopenssl +LIBCURL_GNUTLS:libgnutls +LIBCURL_MBEDTLS:libmbedtls
   DEPENDS += +LIBCURL_ZLIB:zlib +LIBCURL_ZSTD:libzstd +LIBCURL_THREADED_RESOLVER:libpthread +LIBCURL_LDAP:libopenldap
-  DEPENDS += +LIBCURL_LIBIDN2:libidn2 +LIBCURL_SSH2:libssh2 +LIBCURL_NGHTTP2:libnghttp2 +ca-bundle
+  DEPENDS += +LIBCURL_LIBIDN2:libidn2 +LIBCURL_SSH2:libssh2 +LIBCURL_NGHTTP2:libnghttp2 +ca-bundle +kmod-sched
```
Comment (stangri) [inline net/curl/Makefile:94]: Sorry, could you please elaborate why kmod-sched needs to be added as a dependency?

**Comment by stangri** [comment]
@PolynomialDivision Nick, please let me know if you no longer have time to finish this and would rather close this PR, so I can create a new one.

**Comment by PolynomialDivision** [comment]
@stangri Go for it. :)

**Pull request closed without merging**